### PR TITLE
chore: Use setup instead of compat render for stubs

### PR DESCRIPTION
**Pull request by freakzlike**
This PR will use the setup function for stubs instead of the render function with compatibility mode. Also some refactoring so transition and teleport stubs don't use separate function

### CODE REVIEW

**Review by cexbrayat — APPROVED** [review]
I like it, thanks @freakzlike 
I'll let @lmiller1990 or @xanf take another look, but LGTM

File: src/stubs.ts [hunk 1]
```diff
@@ -190,8 +169,9 @@ export function stubComponents(
       stubs['transition-group']
     ) {
       return [
-        createTransitionStub({
-          name: 'transition-group-stub'
+        createStub({
+          name: 'transition-group',
+          renderStubDefaultSlot: true
```
Comment (lmiller1990) [inline src/stubs.ts:174]: I guess this will break some shallow snapshots, but probably fine.

**Review by lmiller1990 — APPROVED** [review]
Thanks!

File: src/stubs.ts (hunk 1 above)
Comment (freakzlike) [reply src/stubs.ts:174]: The `-stub` is added inside of `createStub`

https://github.com/vuejs/vue-test-utils-next/blob/cee2c9ccceef04b256968dd0010d924e21b4c000/src/stubs.ts#L70-L76

**Pull request merged**